snuba/subscriptions/scheduler_processing_strategy.py

```diff
@@ -151,3 +165,72 @@ def terminate(self) -> None:
     def join(self, timeout: Optional[float] = None) -> None:
         self.__next_step.close()
         self.__next_step.join(timeout)
+
+
+class ScheduleSubscriptions(ProcessingStrategy[CommittableTick]):
+    """"
+    This strategy is responsible for scheduling all of the subscriptions
+    for a given tick.
+
+    The subscriptions to be scheduled are those that:
+    - Have the same partition index as the tick received, and
+    - Are scheduled to be run within the time interval defined by the tick
+
+    For backward compatibility, this assumes that the number of partitions of
+    the subscription storage matches the number of partitions of the main topic
+    being subscribed to.
+
+    The subscriptions to be scheduled are are encoded and produced to the
+    scheduled topic to be picked up by the subscription executor to run
+    the actual query later.
+
+    If all of the scheduled subscription messages for the tick are successfully
+    produced, and tick specifies `should_commit` = True, then the offset is committed.
+    """
+
+    def __init__(
+        self,
+        entity_key: EntityKey,
+        schedulers: Mapping[int, Scheduler[Subscription]],
+        producer: Producer[KafkaPayload],
+        scheduled_topic_spec: KafkaTopicSpec,
+        commit: Callable[[Mapping[Partition, Position]], None],
+    ) -> None:
+        self.__schedulers = schedulers
+        self.__encoder = SubscriptionScheduledTaskEncoder(entity_key)
+        self.__producer = producer
+        self.__scheduled_topic = Topic(scheduled_topic_spec.topic_name)
+        self.__commit = commit
+        self.__closed = False
+
+    def poll(self) -> None:
+        pass
```

Comment:
It was originally because submit() was waiting for all the futures to finish before. Since we're no longer waiting there, now updated poll() to evict completed futures from the queue and commit offsets.